tests/functional/vault/test_strategies.py

```diff
@@ -30,29 +28,59 @@ def other_strategy(gov, vault, TestStrategy):
 
 
 @pytest.fixture
-def wrong_strategy(gov, Vault, Token, TestStrategy):
-    otherToken = gov.deploy(Token, 18)
+def other_token(gov, Token):
+    yield gov.deploy(Token, 18)
+
+
+@pytest.fixture
+def strategy_with_wrong_vault(gov, token, vault, Vault, TestStrategy):
     otherVault = gov.deploy(Vault)
     otherVault.initialize(
-        otherToken,
+        token,
         gov,
         gov,
-        otherToken.symbol() + " yVault",
-        "yv" + otherToken.symbol(),
+        token.symbol() + " yVault",
+        "yv" + token.symbol(),
         gov,
     )
+    assert otherVault.token() == token
+    assert otherVault != vault
     otherVault.setDepositLimit(2 ** 256 - 1, {"from": gov})
     yield gov.deploy(TestStrategy, otherVault)
 
 
+@pytest.fixture
+def strategy_with_wrong_want_token(gov, vault, other_token, Token, TestStrategy):
+    strategy = gov.deploy(TestStrategy, vault)
+    assert strategy.want() == vault.token()
+    assert strategy.vault() == vault
+    strategy._setWant(other_token)
+    assert strategy.want() == other_token
+    yield strategy
+
+
 def test_addStrategy(
-    chain, gov, vault, strategy, other_strategy, wrong_strategy, rando
+    chain,
+    gov,
+    vault,
+    strategy,
+    other_strategy,
+    strategy_with_wrong_want_token,
+    strategy_with_wrong_vault,
+    rando,
 ):
 
     # Only governance can add a strategy
     with brownie.reverts():
         vault.addStrategy(strategy, 100, 10, 20, 1000, {"from": rando})
 
+    # Can't add a strategy during emergency shutdown
+    vault.setEmergencyShutdown(True, {"from": gov})
+    with brownie.reverts():
+        vault.addStrategy(strategy, 100, 10, 20, 1000, {"from": gov})
+    chain.undo()
+    chain.undo()
```

Comment:
Why undo twice? this puts it before the test?

Reply:
I can't find any documentation on this, but as far as I can tell from experimentation - at line 81, the `addStrategy` call from line 80 is still in the transaction history. That's why I added the second undo.

Reply:
I don't think that is the case, but if you were having issues with it we should at least add a comment here as to what is being rolled back with each call.

Reply:
@snorkr4 what would you suggest for a comment here?

Reply:
@snorkr4 nvm, saw your suggestion and merged. thanks!